src/dpso/Particle.cpp

```diff
@@ -73,3 +73,24 @@ void Particle::calculate_velocity(EdgesSet global_best_position, const DPSOConfi
     for (const auto &e : particle_best_position)
         velocity.insert(e * config.cognitive_coefficient * random(generator));
 }
+
+void Particle::calculate_new_position(const DPSOConfig &config) {
+    std::random_device random_device;
+    std::mt19937 generator(random_device());
+    std::uniform_real_distribution<> random(0.0, 1.0);
+
+    EdgesSet new_position{};
+
+    // Filter edges from velocity
+    for (const auto &e : velocity) {
+        if (random(generator) <= e.propability)
+            new_position.insert(e);
+    }
+
+    // Filter edges from current position
+    for (const auto &e : position) {
+        if (random(generator) <= random(generator) * config.previous_pos_impact_coefficient)
```

Comment:
wzor ten jest zgodny z tym co jest zaprezentowane w alg3,
ale jak na to patrze to sie zastanawiam czy powinno jeszcze po prawej stronie byc pomnozone przez e.propability ?
```suggestion
        if (random(generator) <= random(generator) * config.previous_pos_impact_coefficient * e.propability)
```

Reply:
Też się nad tym zastanawiałem. Możliwe, że tak, ale okaże się przy testach całego algorytmu.

Reply:
dodaje to do TODO zeby pamietac